### ISSUE TITLE
(rbi fix) A block in CSV#foreach gets CSV::Row type value as formal argument not only Array

### ISSUE THREAD

**Issue by owlworks** [sorbet/sorbet#1579]
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0Arequire%20'csv'%0ACSV.foreach('sample.csv'%2C%20headers%3A%20%3Afirst_row)%20do%20%7Crow%7C%0A%20%20puts%20row.class%0A%20%20puts%20row%5B%22id%22%5D%0Aend)

```ruby
# typed: true
require 'csv'
CSV.foreach('sample.csv', headers: :first_row) do |row|
  puts row.class
  puts row["id"]
end
```

Execution output (sample)
```
CSV::Row
1
```

#### Observed output

```
editor.rb:5: Expected T.any(Integer, Float) but found String("id") for argument arg0 https://srb.help/7002
     5 |  puts row["id"]
               ^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/master/rbi/core/array.rbi#L71: Method Array#[] has specified arg0 as T.any(Integer, Float)
    71 |        arg0: T.any(Integer, Float),
                ^^^^
  Got String("id") originating from:
    editor.rb:5:
     5 |  puts row["id"]
                   ^^^^
Errors: 1
```

#### Expected behavior

I expect to not raise any error. CSV#foreach method with some options gives CSV::Row type value, not Array type value.

Therefore I would like to try rewrite rbi of CSV as Union type.

https://github.com/sorbet/sorbet/blob/31878612b8762643084cc15ee1648d9061235666/rbi/stdlib/csv.rbi#L16

```ruby
blk: T.proc.params(arg0: T.any(T::Array[T.nilable(String)], CSV::Row)).void,
```

Thanks you.

**Comment by jez** [comment]
https://sorbet.org/docs/faq#it-looks-like-sorbet-s-types-for-the-stdlib-are-wrong

**Comment by Morriar** [comment]
The proposed solution was merged in #3122.

It does not entirely solves the issue as you still need to cast `row` as a `CSV::Row` but I think this is the closer we can get with Sorbet.

**Comment by jez** [comment]
Thanks!
